src/c++/perf_analyzer/test_command_line_parser.cc

```diff
@@ -201,51 +202,52 @@ CHECK_PARAMS(PAParamsPtr act, PAParamsPtr exp)
-                                                                           \
-    REQUIRE_NOTHROW(act = parser.Parse(argc, argv));                       \
-    CHECK(!parser.UsageCalled());                                          \
-                                                                           \
-    exp_val = 29;                                                          \
-  }                                                                        \
-                                                                           \
-  SUBCASE("missing value")                                                 \
-  {                                                                        \
-    int argc = 4;                                                          \
-    char* argv[argc] = {app_name, "-m", model_name, option_name};          \
-                                                                           \
-    opterr = 0;                                                            \
-    REQUIRE_NOTHROW(act = parser.Parse(argc, argv));                       \
-    CHECK(parser.UsageCalled());                                           \
-    CHECK_STRING("Usage Message", parser.GetUsageMessage(), "");           \
+#define CHECK_INT_OPTION(option_name, exp_val, msg)                          \
+  SUBCASE("valid value")                                                     \
+  {                                                                          \
+    int argc = 5;                                                            \
+    char* argv[argc] = {app_name, "-m", model_name, option_name, "2000"};    \
+    CAPTURE(argv[3]);                                                        \
+    CAPTURE(argv[4]);                                                        \
+                                                                             \
+    REQUIRE_NOTHROW(act = parser.Parse(argc, argv));                         \
+    CHECK(!parser.UsageCalled());                                            \
+    CAPTURE(parser.GetUsageMessage());                                       \
+                                                                             \
+    exp_val = 2000;                                                          \
+    CAPTURE(exp_val);                                                        \
+  }                                                                          \
+                                                                             \
+  SUBCASE("negative value")                                                  \
+  {                                                                          \
+    int argc = 5;                                                            \
+    char* argv[argc] = {app_name, "-m", model_name, option_name, "-2000"};   \
+    REQUIRE_THROWS_WITH_AS(                                                  \
```

Comment:
I think it would be better to use `CHECK_THROWS_WITH_AS()` so that subsequent subcases can still execute and we can see other potential bugs from a single run of the unit test.

Reply:
Changed to `CHECK_THROWS_WITH_AS`